apps/backend/src/donations/donations.controller.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  UseGuards,
+  Query,
+} from '@nestjs/common';
+import { DonationsService } from './donations.service';
+import { CreateDonationDto } from './dto/create-donation.dto';
+import { UpdateDonationDto } from './dto/update-donation.dto';
+import { FilterDonationsDto } from './dto/filter-donations.dto';
+import { AuthGuard } from '@nestjs/passport';
+import { Logger } from '@nestjs/common';
+import { ValidationPipe } from '@nestjs/common';
+
+@Controller('donations')
+export class DonationsController {
+  constructor(private readonly donationsService: DonationsService) {}
+
+  @Get('orders')
+  filter(@Query(new ValidationPipe()) filterDonationsDto: FilterDonationsDto) {
+    return this.donationsService.filter(filterDonationsDto);
+  }
+
+  @Post()
+  create(@Body() createDonationDto: CreateDonationDto) {
+    return this.donationsService.create(createDonationDto);
+  }
+
+  @Get()
+  findAll() {
+    return this.donationsService.findAll();
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    return this.donationsService.findOne(+id);
```

Comment:
what does +id do?

Reply:
converts to number representation of id
https://en.wikibooks.org/wiki/JavaScript/Operators#Arithmetic_operators